pkg/operator/configobservation/auth/auth_metadata.go

```diff
@@ -0,0 +1,112 @@
+package auth
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/api/config/v1"
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/configobservation"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+const (
+	targetNamespaceName   = "openshift-kube-apiserver"
+	oauthMetadataFilePath = "/etc/kubernetes/static-pod-resources/configmaps/oauth-metadata/oauthMetadata"
+	configNamespace       = "openshift-config"
+	managedNamespace      = "openshift-config-managed"
+)
+
+// ObserveAuthMetadata fills in authConfig.OauthMetadataFile with the path for a configMap referenced by the authentication
+// config.
+func ObserveAuthMetadata(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
```

Comment:
The `ObserveAuthMetadata` func looks fine to me.  I will defer the specifics of the rest of this PR to @openshift/sig-master (the signature for the observe func does look like it is getting out of hand).

Reply:
> The `ObserveAuthMetadata` func looks fine to me. I will defer the specifics of the rest of this PR to @openshift/sig-master (the signature for the observe func does look like it is getting out of hand).

Please be specific.

You need listers, you need an event recorder, you need the existingconfig.  Using a struct while we settle what this looks like in 4.0 will allow code to keep "working" while not understanding semantic differences in behavior. 
 Comments like this without a viable alternative proposed are simply noise

Reply:
> The `ObserveAuthMetadata` func looks fine to me. I will defer the specifics of the rest of this PR to @openshift/sig-master (the signature for the observe func does look like it is getting out of hand).

No.  @openshift/sig-auth is the owner of this function.  If it causes rippling failures the master team will disable it, but the ownership belongs to the team providing the config.